app/views/tasks_views.py

```diff
@@ -1,48 +1,51 @@
-from flask import Blueprint, jsonify, request
-from ..db_services.tasks_service import get_all_tasks, create_task, get_task_by_id, update_task, delete_task,get_user_tasks
-from ..route.tasks import tasks_blueprint
 from flask_login import current_user
+from flask import jsonify, request
+from flask_jwt_extended import jwt_required,get_jwt_identity
+from ..db_services.tasks_service import get_all_tasks, create_task, get_task_by_id, update_task, delete_task
+from ..route.tasks import tasks_blueprint
 
 
 @tasks_blueprint.route('/all', methods=["GET"])
+@jwt_required()
 def manage_tasks():
-    if request.method == "GET":
-        tasks_list = get_all_tasks()
-        return jsonify(tasks_list=tasks_list)
+        current_user_id = get_jwt_identity()
+        if current_user_id:
+            tasks_list = get_all_tasks()
+            return jsonify(tasks_list=tasks_list)
+
+
 @tasks_blueprint.route('/create', methods=["POST"])
+@jwt_required()
 def creats_tasks():
-        try:          
+        try:
             data = request.get_json()           
-            user_id = data.get('user_id')          
+            user_id = data.get('user_id')   
             result = create_task(title=data['title'], description=data['description'], user_id = current_user.id)
             return jsonify(result)
         except Exception as e:
             print(str(e))
             return jsonify(error="give data in JSON format in 'in raw'")
 
-@tasks_blueprint.route('/<string:task_id>', methods=["GET", "PUT", "DELETE"])
-def manage_task(task_id):
-    if request.method == "GET":
-        task = get_task_by_id(task_id)
-        if task:
-            return jsonify(task=task)
-        else:
-            return jsonify(error=f"No task found with id {task_id}.")
 
-    elif request.method == "PUT":
-        try:
-            data = request.get_json()
-            result = update_task(task_id, description=data.get('description'))
+@tasks_blueprint.route('task/<string:task_id>', methods=["GET", "PUT", "DELETE"])
+@jwt_required()
+def manage_task(task_id):
+    current_user_id = get_jwt_identity()
+    if current_user_id:
+        if request.method == "GET":
+            task = get_task_by_id(task_id)
```

Comment:
You should be getting task_id along with user_id until the role is employee.
If it's admin, then he can fetch any task.
If it's manager then he only fetch the tasks to which he is the assignee.